lib/Doctrine/DBAL/Driver/Statement.php

```diff
@@ -43,12 +45,12 @@
      *                       this will be a parameter name of the form :name. For a prepared statement
      *                       using question mark placeholders, this will be the 1-indexed position of the parameter.
      * @param mixed   $value The value to bind to the parameter.
-     * @param integer $type  Explicit data type for the parameter using the PDO::PARAM_* constants.
+     * @param integer $type  Explicit data type for the parameter using the {@link \Doctrine\DBAL\ParameterType}
+     *                       constants.
      *
      * @return boolean TRUE on success or FALSE on failure.
      */
-    function bindValue($param, $value, $type = null);
-
+    public function bindValue($param, $value, $type = ParameterType::STRING);
```

Comment:
Isn't change of default here from null to string BC breaking? (Applies to other signature changes as well.)

Reply:
PHP is fine with that judging by https://3v4l.org/95JYL. Semantically, both `null` and `ParameterType::STRING` are the same since the latter is default everywhere.

Reply:
I'd be cautious with that. Does the test suite at least contain a test with `bindValue()` called only with 2 parameters?

Reply:
@Ocramius there is a test like that:

https://github.com/doctrine/dbal/blob/bc23831907cdb0bc462b2460eb31a21d021ac6c2/tests/Doctrine/Tests/DBAL/Functional/StatementTest.php#L180

This change indeed was only required for `develop` to make this code work:
https://github.com/doctrine/dbal/blob/3fd923919fc6bc81661c30c086b97f58289a38f8/lib/Doctrine/DBAL/Driver/PDOStatement.php#L73

I understand your concern but I don't know what to look for. I manually checked all implementations, and the all either default to binding as string explicitly (IBM DB2, MySQLi, PDO) or don't specify the string/null type at all (OCI8, SQL Server) or seem to not even accept `NULL` (SQLAnywhere).

Furthermore, `NULL` appears to not be a valid type for PDO and logically makes little sense as well:
```php
$conn = new PDO('mysql:host=localhost;dbname=mysql');
$stmt = $conn->prepare('SELECT ?');
$stmt->bindValue(1, 'foo', null);
$stmt->execute();
$value = $stmt->fetchColumn();
var_dump($value);

// NULL
```
I can revert changes in all `Statement::bind*()` signatures if needed.

Reply:
I'm fine if a test already exists and doesn't explicitly pass the third argument to `bindValue()`.

My only issue was that without a test verifying the default parameter we'd have a stability problem :-)

And yes, PDO is (as usual) being weird with default values :-P